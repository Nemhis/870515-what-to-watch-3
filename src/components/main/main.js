import React from 'react';
import PropTypes from 'prop-types';

const Main = ({films}) => {
  const filmsList = films.map((filmName, index) => (
    <article key={index} className="small-movie-card catalog__movies-card">
      <div className="small-movie-card__image">
        <img src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg" alt={filmName} width="280" height="175"/>
      </div>
      <h3 className="small-movie-card__title">
        <a className="small-movie-card__link" href="#">{filmName}</a>
      </h3>
    </article>
  ));

  return <section className="catalog">
    <h2 className="catalog__title visually-hidden">Catalog</h2>

    <div className="catalog__movies-list">{filmsList}</div>
  </section>;
};

Main.propTypes = {
  films: PropTypes.arrayOf(PropTypes.string)
};

export default Main;
