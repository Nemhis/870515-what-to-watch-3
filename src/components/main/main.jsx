import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import FilmsList from '../films-list/films-list.jsx';
import GenresList from '../genres-list/genres-list.jsx';
import ShowMoreButton from '../show-more-button/show-more-button.jsx';
import PromoFilmCard from '../promo-film-card/promo-film-card.jsx';

const Main = ({films}) => {
  return (
    <React.Fragment>
      <PromoFilmCard film={films[0]}/>

      <div className="page-content">
        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <GenresList />

          <FilmsList films={films} />

          <ShowMoreButton />
        </section>
      </div>
    </React.Fragment>
  );
};

Main.propTypes = {
  films: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const mapStateToProps = ({shownFilms}) => ({
  films: shownFilms,
});

export {Main};
export default connect(mapStateToProps)(Main);
