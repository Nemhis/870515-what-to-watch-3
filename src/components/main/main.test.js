import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import {Main} from './main.jsx';

const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
    genre: `Comedy`,
    released: 2014,
    previewImg: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    name: `Bohemian Rhapsody`,
    previewImg: `img/bohemian-rhapsody.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    name: `Macbeth`,
    previewImg: `img/macbeth.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];
const genres = [`Genre 1`, `Genre 2`];

it(`Render correctly <Main>`, () => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(<Main films={films} genres={genres} onSelectGenre={() => {}} />);

  expect(tree).toMatchSnapshot();
});
