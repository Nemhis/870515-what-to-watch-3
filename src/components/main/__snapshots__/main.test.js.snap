// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly <Main> 1`] = `
<React.Fragment>
  <Memo(Connect(PromoFilmCard))
    film={
      Object {
        "backgroundImg": "img/bg-the-grand-budapest-hotel.jpg",
        "genre": "Comedy",
        "name": "Fantastic Beasts: The Crimes of Grindelwald",
        "posterImg": "img/the-grand-budapest-hotel-poster.jpg",
        "previewImg": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "released": 2014,
      }
    }
  />
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <Memo(Connect(GenresList)) />
      <Memo(Connect(FilmsList))
        films={
          Array [
            Object {
              "backgroundImg": "img/bg-the-grand-budapest-hotel.jpg",
              "genre": "Comedy",
              "name": "Fantastic Beasts: The Crimes of Grindelwald",
              "posterImg": "img/the-grand-budapest-hotel-poster.jpg",
              "previewImg": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
              "released": 2014,
            },
            Object {
              "name": "Bohemian Rhapsody",
              "previewImg": "img/bohemian-rhapsody.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            },
            Object {
              "name": "Macbeth",
              "previewImg": "img/macbeth.jpg",
              "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
            },
          ]
        }
      />
      <Memo(Connect(ShowMoreButton)) />
    </section>
  </div>
</React.Fragment>
`;
